src/clp_ffi_js/ir/StreamReader.cpp

```diff
@@ -48,204 +33,40 @@ auto StreamReader::create(DataArrayTsType const& data_array) -> StreamReader {
-            break;
-        }
-        message = parsed.value();
 
-        m_ts_pattern.insert_formatted_timestamp(log_event.get_timestamp(), message);
-
-        EM_ASM(
-                { Emval.toValue($0).push([UTF8ToString($1), $2, $3, $4]); },
-                results.as_handle(),
-                message.c_str(),
-                log_event.get_timestamp(),
-                log_event.get_log_level(),
-                log_event_idx + 1
+        return std::unique_ptr<IrStreamReader>(
+                new IrStreamReader(std::move(stream_reader_data_context))
         );
     }
+    SPDLOG_CRITICAL("Unable to create reader for CLP stream with version {}.", version);
 
-    return DecodedResultsTsType(results);
-}
-
-StreamReader::StreamReader(
-        StreamReaderDataContext<four_byte_encoded_variable_t>&& stream_reader_data_context
-)
-        : m_stream_reader_data_context{std::make_unique<
-                  StreamReaderDataContext<four_byte_encoded_variable_t>>(
-                  std::move(stream_reader_data_context)
-          )},
-          m_ts_pattern{m_stream_reader_data_context->get_deserializer().get_timestamp_pattern()} {}
-
-auto StreamReader::create_data_context(
-        std::unique_ptr<clp::streaming_compression::zstd::Decompressor>&& zstd_decompressor,
-        clp::Array<char> data_buffer
-) -> StreamReaderDataContext<four_byte_encoded_variable_t> {
-    rewind_reader_and_validate_encoding_type(*zstd_decompressor);
-
-    auto result{
-            clp::ir::LogEventDeserializer<four_byte_encoded_variable_t>::create(*zstd_decompressor)
+    throw ClpFfiJsException{
+            clp::ErrorCode::ErrorCode_Unsupported,
+            __FILENAME__,
+            __LINE__,
+            std::format("Unable to create reader for CLP stream with version {}.", version)
     };
-    if (result.has_error()) {
-        auto const error_code{result.error()};
-        SPDLOG_CRITICAL(
-                "Failed to create deserializer: {}:{}",
-                error_code.category().name(),
-                error_code.message()
-        );
-        throw ClpFfiJsException{
-                clp::ErrorCode::ErrorCode_Failure,
-                __FILENAME__,
-                __LINE__,
-                "Failed to create deserializer"
-        };
-    }
-
-    return {std::move(data_buffer), std::move(zstd_decompressor), std::move(result.value())};
 }
 }  // namespace clp_ffi_js::ir
 
 namespace {
-EMSCRIPTEN_BINDINGS(ClpIrStreamReader) {
+EMSCRIPTEN_BINDINGS(ClpStreamReader) {
+    // Output to JS types
     emscripten::register_type<clp_ffi_js::ir::DataArrayTsType>("Uint8Array");
     emscripten::register_type<clp_ffi_js::ir::DecodedResultsTsType>(
             "Array<[string, number, number, number]>"
     );
     emscripten::register_type<clp_ffi_js::ir::FilteredLogEventMapTsType>("number[] | null");
-
-    emscripten::class_<clp_ffi_js::ir::StreamReader>("ClpIrStreamReader")
+    // Input from JS types
```

Comment:
```suggestion

    // JS types used as outputs
```

One thing I've picked up from Junhao's eslint rules is ensuring there's a newline before comments.